command/agent/agent_auto_auth_self_heal_test.go

```diff
@@ -353,25 +319,28 @@ func Test_NoAutoAuthSelfHealing_BadPolicy(t *testing.T) {
 		ExitAfterAuth: false,
 	}
 
+	pathTemplateDestination := makeTempFile(t, "kv-data", "")
+	fileInfo, err = os.Stat(pathTemplateDestination)
+	require.NoError(t, err)
+	templateDestModTime := fileInfo.ModTime()
 	templateTest := &ctconfig.TemplateConfig{
-		Contents:    pointerutil.StringPtr(kvDataTemplateContents),
-		Destination: pointerutil.StringPtr(pathKVData),
+		Contents:    pointerutil.StringPtr(`"{{ with secret "secret/data/otherapp" }}{{ .Data.data.username }}{{ end }}"`),
+		Destination: pointerutil.StringPtr(pathTemplateDestination),
 	}
 	templatesToRender := []*ctconfig.TemplateConfig{templateTest}
 
-	var server *template.Server
-	server = template.NewServer(&sc)
+	server := template.NewServer(&sc)
 	go func() {
 		errCh <- server.Run(ctx, ah.TemplateTokenCh, templatesToRender, ah.AuthInProgress, ah.InvalidToken)
 	}()
 
-	// Trigger template render (mark the time as being earlier, based on the render interval)
-	preTriggerTime := time.Now().Add(-secretRenderInterval)
+	// Send token to the template channel, and wait for the sink to be
+	// populated.
 	ah.TemplateTokenCh <- token
-	_, err = waitForFiles(t, pathTokenFile, preTriggerTime)
+	_, err = waitForFiles(t, pathSinkFile, sinkFileModifiedTime)
```

Comment:
This was the root cause - the pre-trigger time might be before the sink we created earlier. Instead, we're waiting for the sink to be updated past the sink's original time